app/src/test/java/org/simple/clinic/sync/ModelSyncTest.kt

```diff
@@ -0,0 +1,67 @@
+package org.simple.clinic.sync
+
+import com.google.common.truth.Truth.assertThat
+import com.nhaarman.mockito_kotlin.any
+import com.nhaarman.mockito_kotlin.mock
+import com.nhaarman.mockito_kotlin.whenever
+import io.reactivex.Completable
+import junitparams.JUnitParamsRunner
+import junitparams.Parameters
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.simple.clinic.patient.sync.PatientSync
+
+@RunWith(JUnitParamsRunner::class)
+class ModelSyncTest {
```

Comment:
Good question. I don't have a clear answer at the moment.

It seemed easier to do this as a composite test right now for `ModelSyncTest`. I could also take a stab at converting `BaseDataSyncAndroidTest` to a composite form to see if that would be cleaner.

Reply:
Actually, with the introduction of the `ModelSync` interface which describes a common interface all synceable models, it is possible to make it composite. Earlier, even though the individual instances of Sync classes have an identical api, we couldn't abstract them in tests since they were not conforming to a single type.

Reply:
Decision made offline was to change `BaseDataSyncAndroidTest` (now `BaseSyncCoordinatorAndroidTest`) after implementing the `ModelSync` interface on all the Sync classes to a composed form instead of inherited.